projects/whydense/cifar-100/cifar_experiment.py

```diff
@@ -0,0 +1,529 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+# Original Code here:
+# https://github.com/pytorch/examples/blob/master/mnist/main.py
+
+import time
+import math
+import random
+import numpy as np
+
+import torch
+import torch.nn as nn
+from nupic.torch.modules import (
+  SparseWeights, SparseWeights2d,
+  Flatten, KWinners2d, KWinners,
+  rezeroWeights, updateBoostStrength)
+from torchvision import datasets, transforms
+from torch.utils.data import DataLoader
+
+from nupic.research.frameworks.pytorch.model_utils import *
+from nupic.research.frameworks.pytorch.image_transforms import *
+
+from collections import deque
+
+# monkey patch ReduceLROnPlateau, no get_lr implemented 
+def get_lr(self):
+    for param_group in self.optimizer.param_groups:
+        return param_group['lr']
+torch.optim.lr_scheduler.ReduceLROnPlateau.get_lr = get_lr
+
+def CNNSize(width, kernel_size, padding=1, stride=1):
+  return (width - kernel_size + 2 * padding) / stride + 1
+
+def create_test_loaders(dataset, noise_values, batch_size, data_dir):
+  """
+  Create a list of data loaders, one for each noise value
+  """
+  print("Creating test loaders for noise values:", noise_values)
+  loaders = []
+  for noise in noise_values:
+
+    transform_noise_test = transforms.Compose([
+      transforms.ToTensor(),
+      transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
```

Comment:
Did you verify this was the correct normalization for CIFAR-100? I took it from a CIFAR-10 sample, but not sure whether it should be the same for CIFAR100.